### DISCUSSION TITLE
Creating a random point in a circle

### DISCUSSION THREAD

**Discussion by cscetbon** [Turfjs/turf#2542] [General]
I'm trying to use the following code to generate a random point that is included in a circle, but when I calculate the distance from the center of the circle to this point it's exceeds the radius from far, what am I doing wrong ?

```
const turf = require('@turf/turf')

const center = [41.3843, -92.6767]
const radius = 5

var randomPoints = turf.randomPoint(1, {
    bbox: turf.bbox(turf.circle(center, radius)),
})

var randomPoint = randomPoints.features[0]
const distance = turf.distance(turf.point(center), randomPoint)

console.log(`Distance: ${Math.round(distance * 100) / 100} kilometers`)
```

and I get something like `Distance: 594 kilometers`

**Comment by twelch** [comment]
I believe this function only supports creating random points within the bounding box you pass, which you appear to pass the bounding box of your circle. The bounding box of your circle is the smallest rectangle that fits your circle,which will include areas outside of the circle itself.

I don't have an elegant solution to do what you want in turf offhand, but brute force you could keep creating random points until you get one 'within' your circle or with distance from center less than or equal to the radius.

**Reply by cscetbon** [reply]
How can the smallest rectangle that fits for my circle contain a point which is as far as 594 kms from the center when my radius is only 5 kms ?

**Reply by twelch** [reply]
Ahh I didn't notice that important detail, apologies. I'm on my phone and just had a minute.

So the final result is odd. Can you share what the random point generated was for this test case? Is it always or just sometimes incorrect? And if you have time what is the earliest point in your series of function calls that the result seemed incorrect? Is it the random point? The distance calculation? The circle? Any additional detail is appreciated and will help someone contributing their time to troubleshoot.

**Reply by smallsaucepan** [reply]
@cscetbon it might be that your latitude (-92.6767) is past the South Pole. Results will be unpredictable if you go outside the limits of what Turf expects. Longitude -180 to 180, and latitude -90 to 90.

**Reply by cscetbon** [reply]
Thanks @smallsaucepan it seems to be a good explanation !

**Reply by smallsaucepan** [reply]
Happy to help. Btw, if within the bounding box as Tim suggested doesn't suit, you could try generating a random distance (0-5) and random bearing (0-360) and feeding those into turf-destination with your circle center as the origin.

**Reply by mandano** [reply]
thx for pointing out the approach of random radius, random bearing inserted to destination(); github pilot proposed: create circle -> get bbox from circle -> pass to randomPoint...
